oak-run-elastic/src/main/java/org/apache/jackrabbit/oak/index/ElasticIndexCommand.java

```diff
@@ -216,33 +207,25 @@ private void reindex(ElasticIndexOptions indexOpts, IndexHelper indexHelper, Str
                     indexOpts.getElasticScheme(), indexOpts.getElasticHost(),
                     indexOpts.getElasticPort(), indexOpts.getApiKeyId(), indexOpts.getApiKeySecret())) {
                 indexer.reindex();
-                // Wait for default flush interval before exiting the try block
-                // to make sure the client is not closed before the last flush
-                // TODO : See if this can be handled in a better manner
-                Thread.sleep(ElasticIndexDefinition.BULK_FLUSH_INTERVAL_MS_DEFAULT * 2);
```

Comment:
Great that this is gone!

Reply:
Yes, with the changes in the ElasticBulkProcessHandler in this PR , we wait for ES response acknowledgement for offline reindex (which we weren't doing earlier) - this effectively should ensure that the last BulkProcess is handled before the ElasticConnection is closed here.